pkg/v3/types/interfaces.go

```diff
@@ -1,109 +1,27 @@
-type UpkeepStateUpdater interface {
-	SetUpkeepState(context.Context, CheckResult, UpkeepState) error
-}
+//go:generate mockery --name Coordinator --structname MockCoordinator --srcpkg "github.com/smartcontractkit/chainlink-common/pkg/types/automation" --case underscore --filename coordinator.generated.go
 
-type RetryQueue interface {
-	// Enqueue adds new items to the queue
-	Enqueue(items ...RetryRecord) error
-	// Dequeue returns the next n items in the queue, considering retry time schedules
-	Dequeue(n int) ([]UpkeepPayload, error)
-}
+//go:generate mockery --name MetadataStore --structname MockMetadataStore --srcpkg "github.com/smartcontractkit/chainlink-common/pkg/types/automation" --case underscore --filename metadatastore.generated.go
 
-type ProposalQueue interface {
-	// Enqueue adds new items to the queue
-	Enqueue(items ...CoordinatedBlockProposal) error
-	// Dequeue returns the next n items in the queue, considering retry time schedules
-	Dequeue(t UpkeepType, n int) ([]CoordinatedBlockProposal, error)
-}
-
-//go:generate mockery --name ResultStore --structname MockResultStore --srcpkg "github.com/smartcontractkit/chainlink-automation/pkg/v3/types" --case underscore --filename result_store.generated.go
-type ResultStore interface {
-	Add(...CheckResult)
-	Remove(...string)
-	View() ([]CheckResult, error)
-}
-
-//go:generate mockery --name Coordinator --structname MockCoordinator --srcpkg "github.com/smartcontractkit/chainlink-automation/pkg/v3/types" --case underscore --filename coordinator.generated.go
-type Coordinator interface {
-	PreProcess(_ context.Context, payloads []UpkeepPayload) ([]UpkeepPayload, error)
-
-	Accept(ReportedUpkeep) bool
-	ShouldTransmit(ReportedUpkeep) bool
-	FilterResults([]CheckResult) ([]CheckResult, error)
-	FilterProposals([]CoordinatedBlockProposal) ([]CoordinatedBlockProposal, error)
-}
-
-//go:generate mockery --name MetadataStore --structname MockMetadataStore --srcpkg "github.com/smartcontractkit/chainlink-automation/pkg/v3/types" --case underscore --filename metadatastore.generated.go
-type MetadataStore interface {
-	SetBlockHistory(BlockHistory)
-	GetBlockHistory() BlockHistory
-
-	AddProposals(proposals ...CoordinatedBlockProposal)
-	ViewProposals(utype UpkeepType) []CoordinatedBlockProposal
-	RemoveProposals(proposals ...CoordinatedBlockProposal)
-
-	Start(context.Context) error
-	Close() error
-}
-
-//go:generate mockery --name Ratio --structname MockRatio --srcpkg "github.com/smartcontractkit/chainlink-automation/pkg/v3/types" --case underscore --filename ratio.generated.go
-type Ratio interface {
-	// OfInt should return n out of x such that n/x ~ r (ratio)
-	OfInt(int) int
-}
+//go:generate mockery --name Ratio --structname MockRatio --srcpkg "github.com/smartcontractkit/chainlink-common/pkg/types/automation" --case underscore --filename ratio.generated.go
```

Comment:
We're moving the type definitions into common, but automation cares about the mocks, so we're keeping the mocks in this repo (by referencing the chainlink-common package in the generate command)